### PR TITLE
Automation for External Authorization 

### PR DESCRIPTION
**Pull request by pbajjuri20**
https://istio.io/latest/docs/tasks/security/authorization/authz-custom/

Executed the Manual steps and added the automation test case for Maistra-test-tool.

### CODE REVIEW

File: pkg/tasks/security/authorization/externalauthorization.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+			XB3Traceid string `json:"X-B3-Traceid"`
+			XExtAuthz  string `json:"X-Ext-Authz"`
+		}
+
+		type Response struct {
+			Headers Headers
+		}
+
+		var response Response
+		util.Inspect(err, "Failed to get sleep pod name", "", t)
```
Comment (monroe-dylan-m) [inline pkg/tasks/security/authorization/externalauthorization.go:120]: `err` was last set by line 96 and does not indicate an error getting the sleep pod name (likely just a copy-paste mistake from lines 53-54). The same issue occurs on line 131.

File: pkg/tasks/security/authorization/externalauthorization.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+		response1, err := util.Shell("kubectl exec %s -c sleep -n foo -- %s", sleepPod, allow)
+
+		json.Unmarshal([]byte(response1), &response)
+		if response.Headers.XExtAuthz == "allow" && response.Headers.Host == "httpbin.foo:8000" && response.Headers.Accept == "*/*" {
+			util.Log.Infof("Success, verfication the header with allow server")
+		} else {
+			util.Log.Errorf("Failed, verfication the header with allow server")
+		}
+
+		util.Inspect(err, "Failed to get sleep pod name", "", t)
```
Comment (monroe-dylan-m) [inline pkg/tasks/security/authorization/externalauthorization.go:131]: See my comment on line 120.

**Review by monroe-dylan-m — CHANGES_REQUESTED** [review]
Looks good to me, just noticed a typo in `externalauthorization.go` that should be corrected.

File: pkg/tasks/security/authorization/externalauthorization.go (hunk 2 above)
Comment (pbajjuri20) [reply pkg/tasks/security/authorization/externalauthorization.go:132]: updated the err info

Comment (monroe-dylan-m) [reply pkg/tasks/security/authorization/externalauthorization.go:132]: Can you also remove line 132 (originally line 131)?

Comment (pbajjuri20) [reply pkg/tasks/security/authorization/externalauthorization.go]: Thanks for checking, removed the unnecessary line

**Review by monroe-dylan-m — APPROVED** [review]

**Comment by pbajjuri20** [comment]
@monroe-dylan-m: Thanks for approving this test case. 

I will build and run against Jenkins.

**Pull request merged**